File: 20210722/완주하지못한선수.js
function solution(participant, completion) {
  let obj = {};
  for (const i of participant) {
    if (obj[i] !== undefined) {
      obj[i] += 1;
    } else {
      obj[i] = 1;
    }
  }

  for (const i of completion) {
    obj[i] -= 1;
  }
  for (const i of participant) {
    if (obj[i] !== 0) return i;
  }
}

console.log(
  solution(["mislav", "stanko", "mislav", "ana"], ["stanko", "ana", "mislav"])
);

function solution(participant, completion) {
  let dic = completion.reduce(
    (acc, cur) => ((acc[cur] = acc[cur] ? acc[cur] + 1 : 1), acc),
    {}
  );

  return participant.find((value) => {
    if (dic[value]) {
      dic[value] -= 1;
    } else {
      return true;
    }
  });
}

File: 20210722/완주하지못한선수.py
def solution(participant, completion):
    dic = {}
    for i in participant:
        if dic.get(i) != None:
            dic[i] += 1
        else:
            dic[i] = 1

    for i in completion:
        dic[i] -= 1
    for i in list(dic.keys()):
        if dic[i] != 0:
            return i


print(solution(["mislav", "stanko", "mislav", "ana"],
               ["stanko", "ana", "mislav"]))


# def solution(participant, completion):
#     answer = ''
#     temp = 0
#     dic = {}
#     for part in participant:
#         dic[hash(part)] = part
#         temp += int(hash(part))
#     for com in completion:
#         temp -= hash(com)
#     answer = dic[temp]

#     return answer
